src/ui/public/vislib/visualizations/pie_chart.js

```diff
@@ -244,7 +244,7 @@ export function VislibVisualizationsPieChartProvider(Private) {
               x: pos[0],
               y: pos[1],
               left: midAngle < Math.PI ? pos[0] : pos[0] - bbox.width,
-              right: midAngle > Math.PI ? pos[0] + bbox.width : pos[0],
+              right: midAngle < Math.PI ? pos[0] + bbox.width : pos[0],
```

Comment:
this fix doesn't help much. It hide the label which conflict with other but what it should do is to try to find a better place to be shown. Usually there is still plenty of space